install.sh

```diff
@@ -84,85 +63,125 @@ download_mkectl() {
  curl -s -L -o /tmp/mkectl.tar.gz "$DOWNLOAD_URL"
 
  # Verify the file is a valid gzip archive
- if [ -s /tmp/mkectl.tar.gz ] && file /tmp/mkectl.tar.gz | grep -q 'gzip compressed data'; then
+ if [ -s /tmp/mkectl.tar.gz ]; then
    # Extract the downloaded file
-   tar -xvzf /tmp/mkectl.tar.gz -C "$installPath"
-   echo "mkectl is now executable in $installPath"
+   tar -xvzf /tmp/mkectl.tar.gz -C "$installPath" && echo "mkectl is now executable in $installPath" || echo "Error: Downloaded file is not a valid gzip archive" >&2 && exit 1
  else
-   echo "Error: Downloaded file is empty or not a valid gzip archive." >&2
+   echo "Error: Downloaded file is empty." >&2
    exit 1
  fi
 }
 
-main() {
+# Download dependencies for MKE version 4.0.0
+download_dependencies() {
+    printf "\n\n"
 
-  uname="$(detect_uname)"
-  arch="$(detect_arch)"
+    echo "Dependency 1/2 : Install k0sctl"
+    echo "#########################"
 
-  printf "\n\n"
+    if [ -z "${K0SCTL_VERSION}" ]; then
+      echo "Using default k0sctl version 0.19.4"
+      K0SCTL_VERSION=0.19.4
+    fi
 
-  echo "Step 1/3 : Install k0sctl"
-  echo "#########################"
+    k0sctlBinary=k0sctl
+    k0sctlDownloadUrl="$(download_k0sctl_url)"
 
-  if [ -z "${K0SCTL_VERSION}" ]; then
-    echo "Using default k0sctl version 0.19.4"
-    K0SCTL_VERSION=0.19.4
-  fi
 
-  k0sctlBinary=k0sctl
-  installPath=/usr/local/bin
-  k0sctlDownloadUrl="$(download_k0sctl_url)"
+    echo "Downloading k0sctl from URL: $k0sctlDownloadUrl"
+    curl -sSLf "$k0sctlDownloadUrl" >"$installPath/$k0sctlBinary"
 
+    sudo chmod 755 "$installPath/$k0sctlBinary"
+    echo "k0sctl is now executable in $installPath"
+}
 
-  echo "Downloading k0sctl from URL: $k0sctlDownloadUrl"
-  curl -sSLf "$k0sctlDownloadUrl" >"$installPath/$k0sctlBinary"
-
-  sudo chmod 755 "$installPath/$k0sctlBinary"
-  echo "k0sctl is now executable in $installPath"
+# compares versions by ordering them
+# Do not use it to compare any semver formats, as it compares release and pre-release versions incorrectly.
+# It's used only to compare v4.0.1-* pre-release versions
+version_ge() { test "$(echo "$@" | tr " " "\n" | sort -rV | head -n 1)" == "$1"; }
 
-  printf "\n\n"
-  echo "Step 2/3 : Install kubectl"
-  echo "#########################"
+normalizeSemverZero() {
```

Comment:
nit: Mostly for consistency, most of your functions have a short doc header. It'd be nice to summarize these for when we forget what they are doing in the future.

Plus bash just isn't the prettiest language so the next person to work on this will thank you